featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -51,17 +53,49 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
     """
     if not HAS_AZURE_MONITOR_EVENTS_EXTENSION:
         return
-    event = {}
-    if evaluation_event.feature:
-        event[FEATURE_NAME] = evaluation_event.feature.name
+    event: Dict[str, Optional[str]] = {}
+    if not evaluation_event.feature:
+        return
+    event[FEATURE_NAME] = evaluation_event.feature.name
     event[ENABLED] = str(evaluation_event.enabled)
+    event["Version"] = EVALUATION_EVENT_VERSION
 
+    # VariantAllocationPercentage
     if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
         if evaluation_event.variant:
             event[VARIANT] = evaluation_event.variant.name
         event[REASON] = evaluation_event.reason.value
 
-    if evaluation_event.feature and evaluation_event.feature.telemetry:
+        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
+            allocation_percentage = 0
+
+            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
+                for allocation in evaluation_event.feature.allocation.percentile:
+                    if (
+                        evaluation_event.variant
+                        and allocation.variant == evaluation_event.variant.name
+                        and allocation.percentile_to
+                    ):
+                        allocation_percentage += allocation.percentile_to - allocation.percentile_from
+
+            event["VariantAssignmentPercentage"] = str(100 - allocation_percentage)
+        elif evaluation_event.reason == VariantAssignmentReason.PERCENTILE:
+            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
+                allocation_percentage = 0
+                for allocation in evaluation_event.feature.allocation.percentile:
+                    if (
+                        evaluation_event.variant
```

Comment:
It would be better to check whether evaluation_event.variant is None in advance.

Reply:
Is it a possible case that variantAssignmentReason is not None but variant is None? I cannot imagine it will happen.

Reply:
@zhiyuanliang-ms, I've made some changes already to this in my already existing PR that has some updates. https://github.com/microsoft/FeatureManagement-Python/pull/45